src/app/services/journey.service.ts

```diff
@@ -15,52 +16,53 @@ export class JourneyService {
 
   /**
    * search by id
-   * @param value
+   * @param id searched
    */
-  searchById(value: number): Observable<Journey> {
-
+  searchById(id: number): Observable<Journey> {
     const source$ = this.apollo.query({
       query: gql`
         query find{
-          findJourneyById(id: "${value}") {
+          findJourneyById(id: "${id}") {
             id,
             destination
           }
         }
       `
     }).pipe(shareReplay(1));
 
-    const destination$: Observable<any[]> = source$.pipe(map(result => result.data && result.data['findJourneyById']));
+    const destination$: Observable<Journey> = source$.pipe(
+      tap(result => log(result)),
```

Comment:
what is tap ?

Reply:
same as map but with no side effect. Does not change the value.